site/ja/lite/microcontrollers/get_started.md

```diff
@@ -0,0 +1,329 @@
+
+```C++
+#include "tensorflow/lite/micro/examples/hello_world/sine_model_data.h"
+```
+
+### 単体テストを用意する
+
+見ていくコードは単体テストで、それはマイクロコントローラ向けTensorFlow Liteフレームワークの単体テストフレームワークを使います。
+フレームワークを読み込むため、以下のファイルをインクルードします。
+
+```C++
+#include "tensorflow/lite/micro/testing/micro_test.h"
+```
+
+テストは以下のマクロを使って定義されます。
+
+```C++
+TF_LITE_MICRO_TESTS_BEGIN
+
+TF_LITE_MICRO_TEST(LoadModelAndPerformInference) {
+```
+
+コードの残り部分は、モデルの読み込みと推論を実演します。
+
+### ログ取得を用意する
+
+ログ取得の準備するために、`tflite::ErrorReporter` ポインタが作成され、ポインタは `tflite::MicroErrorReporter` インスタンスに対して使用されます。
+
+```C++
+tflite::MicroErrorReporter micro_error_reporter;
+tflite::ErrorReporter* error_reporter = &micro_error_reporter;
+```
+
+この変数はインタープリタに渡され、ログに書くことを許可します。
+マイクロコントローラはしばしばログ取得のさまざまな機構をもつので、`tflite::MicroErrorReporter` の実装は、
+デバイス固有にカスタマイズされて設計されます。
+
+### モデルを読み込む
+
+以下のコードでは、モデルは `char` 配列、つまり `sine_model_data.h` で宣言された `g_sine_model_data` からのデータを使って実体化されます。
+モデルを検査し、そのスキーム・バージョンが我々が使用しているバージョンと互換性があることを確認します。
+
+```C++
+const tflite::Model* model = ::tflite::GetModel(g_sine_model_data);
+if (model->version() != TFLITE_SCHEMA_VERSION) {
+  error_reporter->Report(
+      "与えられたモデルはスキーム・バージョン%dであり、"
+      "サポートされたバージョン%dと一致しません。",
+      model->version(), TFLITE_SCHEMA_VERSION);
+}
+```
+
+### 演算子リゾルバを実体化する
+
+[`AllOpsResolver`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/kernels/all_ops_resolver.h)
+インスタンスが宣言されている。これはインタープリタがモデルで使用されている演算にアクセスするために使用されます。
+
+```C++
+tflite::ops::micro::AllOpsResolver resolver;
+```
+
+`AllOpsResolver` は、マイクロコントローラ向けTensorFlow Liteで利用可能なすべての演算を読み込み、それは多くのメモリを使用します。
+与えられたモデルはこれら演算の一部のみを使用するであろうから、現実のアプリケーションは必要な演算のみを読み込むことが推奨されます。
+
+これは別のクラス、`MicroMutableOpResolver` を使用して実施されます。
+*Micro speech* [`micro_speech_test.cc`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/examples/micro_speech/micro_speech_test.cc) の例で使い方を見ることができます。
+
+### メモリを割り当てる
+
+適当な量のメモリを入力、出力、そして中間配列に事前に割り当てる必要があります。
+これは、`tensor_arena_size` の大きさの `uint8_t` 配列として提供されます。
+
+```C++
+const int tensor_arena_size = 2 * 1024;
+uint8_t tensor_arena[tensor_arena_size];
+```
+
+要求される大きさは使用するモデルに依存し、実験によって決められる必要があるかもしれません。
+
+### インタープリタを実体化する
+
+`tflite::MicroInterpreter` インスタンスを作成し、事前に作成した変数を渡します。
+
+```C++
+tflite::MicroInterpreter interpreter(model, resolver, tensor_arena,
+                                     tensor_arena_size, error_reporter);
+```
+
+### テンソルを割り当てる
+
+インタープリタに対し、 `tensor_arena` からモデルのテンソルにメモリを割り当てるように指示します。
+
+```C++
+interpreter.AllocateTensors();
+```
+
+### 入力の形を検証する
+
+`MicroInterpreter` インスタンスは、`.input(0)` を呼ぶことで、モデルの入力テンソルへのポインタを提供します。
+`0` は最初の（そして唯一の）入力テンソルであることを表します。
+
+```C++
+  // モデルの入力テンソルへポインタを取得する
+  TfLiteTensor* input = interpreter.input(0);
+```
+
+このテンソルを検証し、形と型が期待したものであることを確認します。
+
+```C++
+// 入力は期待するプロパティを持つこと確認する
+TF_LITE_MICRO_EXPECT_NE(nullptr, input);
+// プロパティ "dims" はテンソルの形を教えてくれる。
+// それは次元ごとに1つの要素を持つ。我々の入力は2次元のテンソルで1つの要素を含むので、
+// "dims" は2であるべきである。
```

Comment:
I think your translation has no problem and we have nothing to fix on this line. Since English documents are the "single source of truth", it is better to discuss at the tensorflow/tensorflow repo. I'll do that.

Reply:
https://github.com/tensorflow/tensorflow/issues/37829

Reply:
Sorry, this comment came from my misunderstanding. What I have to check truly is the input shape of `sine_model`, not the input of `model_2 `. Thank you for your suggestion!